core/src/banking_stage.rs

```diff
@@ -2010,16 +1991,11 @@ impl BankingStage {
         slot_metrics_tracker: &mut LeaderSlotMetricsTracker,
     ) -> Result<(), RecvTimeoutError> {
         let mut recv_time = Measure::start("receive_and_buffer_packets_recv");
-        let packet_batches = Self::receive_until(
-            verified_receiver,
-            recv_timeout,
-            Duration::from_millis(MAX_RECEIVE_TIME_MS_PER_ITERATION),
-            MAX_RECEIVE_BATCH_SIZE_PER_ITERATION,
-        )?;
+        let (packet_batches, packet_count) =
+            verified_receiver.recv_timeout(MAX_RECEIVE_PACKETS_PER_ITERATION, recv_timeout)?;
```

Comment:
@pgarg66 - Rather than receiving max of 50k batches, we will now receive max of 100k packets. I believe this change will allow for tighter arbitration (since counting batches is at the mercy of how dense the batches are), but it's not clear if 100k is the right number. Simple bench perf test seems to indicate we haven't completely screwed things up, but curious what your thoughts are on this

Reply:
The newer code (tip of master and v1.10) is actually counting packets now. So the goals of this PR aligns well with what banking_stage is doing today.